astropy/coordinates/sky_coordinate.py

```diff
@@ -13,6 +13,7 @@
 from .. import units as u
 from ..wcs.utils import skycoord_to_pixel, pixel_to_skycoord
 from ..utils.exceptions import AstropyDeprecationWarning
+from ..utils.data_info import DataInfo, BaseInfo
```

Comment:
The names are confusing to me, as they don't make explicit the roles that they have, i.e., one has to know that `DataInfo` is a descriptor, and `BaseInfo` the base for the class passing back information. It doesn't help to call the actual class `SkyCoordDataInfo`. I think I'd go with (in order of strongly to weakly suggested):
* renaming `SkyCoordDataInfo` to `SkyCoordInfo`
* maybe renaming `DataInfo` to `InfoDescriptor` (yes, duplicate but makes point clearer)
* If so, renaming `BaseInfo` to `DataInfo`

I think the code below is clearer if one sees
```
    info = InfoDescriptor(SkyCoordInfo)
```